.golangci-maximal.yml

```diff
@@ -0,0 +1,183 @@
+linters:
+  disable-all: true
+  enable:
+    - copyloopvar
+    - cyclop
+    - depguard
+    - dupword
+    - errcheck
+    - errname
+    - errorlint
+    - exhaustruct
+    - exptostd
+    - funlen
+    - gci
+    - gocheckcompilerdirectives
+    - gocognit
+    - gocritic
+    - gocyclo
+    - godot
+    - goimports
+    - govet
+    - iface
+    - importas
+    - ineffassign
+    - intrange
+    - maintidx
+    - misspell
+    - nestif
+    - nilnesserr
+    - nolintlint
+    - predeclared
+    - recvcheck
+    - revive
+    - staticcheck
+    - testifylint
+    - unconvert
+    - unparam
+    - unused
+    - usestdlibvars
+    - usetesting
+  fast: false
+
+linters-settings:
+  depguard:
+    rules:
+      # This is just an example of a rule you could create, though 
+      forbid-some-pacakge:
+        list-mode: Lax
+        files:
+          - "**/*.go"
+        deny:
+          pkg: "github.com/some/package"
+          desc: "Don't use some/package. It's got issues."
+
+  errcheck:
+    check-type-assertions: true
+
+  gci:
+    sections:
+      - standard
+      - default
+
+  # This configure gocritic so that it is _only_ used to run ruleguard. It's default checks include
+  # some that can be a bit annoying, like insisting on using `switch` instead of if/else all over
+  # the place.
+  gocritic:
+    disable-all: true
+    enabled-checks:
+      - ruleguard
+    settings:
+      ruleguard:
+        rules: "${configDir}/gorules/*.go"
+        # This causes a warning when rules cannot be compiled.
+        failOn: dsl
+
+
+  govet:
+    # This is not part of the default checks, but it's useful for finding bugs and generally
+    # confusing code.
+    shadow: true
+
+  importas:
```

Comment:
I know with proto, we did a lot of aliasing. What exactly are we preventing here?

Reply:
The goal for this is not to prevent the use of aliases. It's to force you to configure the linter so that you always use the _same_ alias whenever you alias an import. With the example config, you can alias `github.com/google/uuid` to `guuid` anywhere or you can import it without an alias. But you can't alias it as `googuuid` in some code and `guuid` in other code.